Bottom: caa79ce9fbeca1b878fccb0e03307a0bbd7a7ff4
Top:    a44a8fd53e868df41ea71b64da13d4a20b05df0f
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 19:24:20 +0000

polygon.c: Temporarily disable coalescing adjacent line segments

We need to extend the tests to properly account for arc segments.


---

diff --git a/src/polygon.c b/src/polygon.c
index fa4531b..9f73d09 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1364,6 +1364,7 @@ RemoveExcessPolygonPoints (LayerType *Layer, PolygonType *Polygon)
   if (Undoing ())
     return (false);
 
+if (0) {
   for (n = 0; n < Polygon->PointN; n++)
     {
       prev = prev_contour_point (Polygon, n);
@@ -1373,6 +1374,7 @@ RemoveExcessPolygonPoints (LayerType *Layer, PolygonType *Polygon)
       line.Point1 = Polygon->Points[prev];
       line.Point2 = Polygon->Points[next];
       line.Thickness = 0;
+#warning THIS TEST IS NOT SUFFICIENT.. DOESNT ACCOUNT FOR DIFFERENT ARC CENTERS
       if (Polygon->Points[prev].included_angle == Polygon->Points[n].included_angle &&
           IsPointOnLine (p->X, p->Y, 0.0, &line))
         {
@@ -1380,6 +1382,7 @@ RemoveExcessPolygonPoints (LayerType *Layer, PolygonType *Polygon)
           changed = true;
         }
     }
+}
   return (changed);
 }
